src/Workspaces/Core/Portable/Workspace/Solution/StateChecksums.cs

```diff
@@ -14,27 +15,43 @@
 
 namespace Microsoft.CodeAnalysis.Serialization;
 
-internal sealed class SolutionStateChecksums(
-    Checksum attributesChecksum,
-    ChecksumCollection projectChecksums,
-    ChecksumCollection analyzerReferenceChecksums,
-    Checksum frozenSourceGeneratedDocumentIdentity,
-    Checksum frozenSourceGeneratedDocumentText) : IChecksummedObject
+internal sealed class SolutionStateChecksums : IChecksummedObject
 {
-    public Checksum Checksum { get; } = Checksum.Create(stackalloc[]
+    public Checksum Checksum { get; }
+
+    public Checksum Attributes { get; }
+    public ImmutableArray<ProjectId> ProjectIds { get; }
+    public ChecksumCollection Projects { get; }
```

Comment:
note: i merge ProejctIds and Projects into a singular collection type in a later PR to keep these more tightly bound